### PR TITLE
Adds support for Kibana 7.10.0

### PR DESCRIPTION
**Pull request by ftianli-amzn**
*Issue #, if available:*

*Description of changes:*

**Version number changes:**
- Bump the plugin version to 1.12.0.0
- Bump Kibana version to 7.10.0
- Bump node version to 10.22.1

**Configuration changes:**
1. kibana.json:
- Change the plugin id to `opendistroAlertingKibana` with camel cases.
- Assign Kibana version in "kibanaVersion" explicitly to avoid inputting Kibana version number manually when running `build` script.
2. package.json
_"kibana":_
- Remove "kibana.version", since the version is defined in "kibana.json" instead.
- Remove "kibana.templateVersion", because neither we are using the templete in that version, nor it's worth to have this field.
_"config":_
- Add a field to set plugin `id` in env parameter which is used for `postbuild` script
_"scripts":_ 
- Update `build` and `start` script according to the [change](https://github.com/elastic/kibana/tree/master/packages/kbn-plugin-helpers#kbnplugin-helpers) of `plugin-helpers`
- Update `eslint` script to ignore all the files defined in `.gitignore` (especially for the new `target` folder)
- Add a `postbuild` script to rename the build artifact to be `<plugin_id>-<plugin_version>.zip` which will be ran after `build`
(reference: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/package.json#L16) 
_"devDependencies":_
- Remove `@kbn/eslint-config-kibana` according to the change of the [usage](https://github.com/elastic/kibana/tree/master/packages/elastic-eslint-config-kibana#elastic-eslint-config-kibana)
- Remove `@kbn/expect` because it shows version conflict with Kibana during bootstrap. Can be used by `import expect from @kbn/expect` instead.
- Remove `@kbn/eslint-import-resolver-kibana` because it's not used, and validated by seeing no difference in the console output of running `eslint`.
_"dependencies":_
- Modify the version ranges of `brace`, `lodash`, `query-string`, `react-router-dom`, and `react-vis` to avoid version conflict with Kibana 7.10.0, where a [new validation](https://github.com/elastic/kibana/blob/e5d8d49164a8d993bdd314dd6c800f9186092148/packages/kbn-pm/src/utils/validate_yarn_lock.ts#L145) is added to force a single version for any dependencies.
3. ".kibana-plugin-helpers.json":
- Update according to the change of the [usage](https://github.com/elastic/kibana/tree/master/packages/kbn-plugin-helpers#kbnplugin-helpers) of `plugin-helpers`
4. ".eslintrc":
- Update configuration file for eslint with reference of [Kibana plugin templete](https://github.com/elastic/kibana/blob/master/packages/kbn-plugin-generator/template/.eslintrc.js.ejs)

**CI changes:**
- Update the test and release workflows to get Kibana version from `kibana.json`

**Test changes:**
- Change the deprecated 'require.requireActual()' call to 'jest.requireActual()'
- Update Jest snapshot files

**Document changes:**
- Update the README file for the `yarn start` running location change, and a link about contributing
- Remove the outdated `TRANSLATION.md`, localization can refer to [kbn-i18n](https://github.com/elastic/kibana/tree/master/packages/kbn-i18n) package
- Update the Release Notes (will do in another PR)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ftianli-amzn** [comment]
Result for `yarn test:jest`
```
Test Suites: 6 skipped, 82 passed, 82 of 88 total
Tests:       20 skipped, 382 passed, 402 total
Snapshots:   135 passed, 135 total
Time:        43.059 s
Ran all test suites.
✨  Done in 44.16s.
```

File: package.json [hunk 1]
```diff
@@ -1,53 +1,47 @@
...
   "scripts": {
     "kbn": "node ../../scripts/kbn",
-    "lint": "../../node_modules/.bin/eslint '**/*.js' -c .eslintrc",
+    "lint": "../../node_modules/.bin/eslint '**/*.js' -c .eslintrc --ignore-path .gitignore",
     "start": "plugin-helpers start",
-    "test:server": "plugin-helpers test:server",
-    "test:browser": "plugin-helpers test:browser",
     "test:jest": "../../node_modules/.bin/jest --config ./test/jest.config.js",
-    "build": "plugin-helpers build"
+    "build": "yarn plugin-helpers build",
```
Comment (dbbaughe) [inline package.json:20]: Why does this need yarn prefixed?

File: package.json [hunk 2]
```diff
@@ -1,53 +1,47 @@
...
   "devDependencies": {
-    "@elastic/eslint-config-kibana": "link:../../packages/eslint-config-kibana",
-    "@elastic/eslint-import-resolver-kibana": "link:../../packages/kbn-eslint-import-resolver-kibana",
-    "@kbn/expect": "link:../../packages/kbn-expect",
-    "@kbn/plugin-helpers": "link:../../packages/kbn-plugin-helpers",
     "husky": "^3.0.0",
     "lint-staged": "^9.2.0"
   },
   "dependencies": {
-    "brace": "^0.11.1",
```
Comment (dbbaughe) [inline package.json:39]: Why the change here?

File: package.json (hunk 1 above)
Comment (ftianli-amzn) [reply package.json:20]: I will have a try without `yarn` prefixed, originally I just followed their instruction https://github.com/elastic/kibana/tree/master/packages/kbn-plugin-helpers#kbnplugin-helpers

File: package.json (hunk 2 above)
Comment (ftianli-amzn) [reply package.json:39]: Kibana seems added a new limitation to have only one version for any dependencies in Kibana. (https://github.com/elastic/kibana/blob/e5d8d49164a8d993bdd314dd6c800f9186092148/packages/kbn-pm/src/utils/validate_yarn_lock.ts#L145)
Without the changes for the version ranges, an error with the above message will occur during `yarn kbn bootstrap`.
I'm not quite sure if the change is correct here.

File: package.json (hunk 1 above)
Comment (ftianli-amzn) [reply package.json:19]: The `plugin-helpers` is not a executable file but a script in the new version, but I'm not very familiar with how the script is defined here, will do more research.

File: package.json [hunk 3]
```diff
@@ -1,53 +1,46 @@
...
+  "version": "1.12.0.0",
   "description": "Kibana Alerting Plugin",
   "main": "index.js",
   "license": "Apache-2.0",
   "homepage": "https://github.com/opendistro-for-elasticsearch/alerting-kibana-plugin",
   "kibana": {
-    "version": "7.9.1",
-    "templateVersion": "6.3.3"
+    "version": "7.10.0"
   },
```
Comment (ohltyler) [inline package.json:10]: I believe you can remove all Kibana-related versioning logic here, since it is maintained in the `kibana.json` manifest file. See [here](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/329)

Comment (ftianli-amzn) [reply package.json:10]: Thanks for reviewing! Yes, you are right, I have tried to remove it once, and it works. 
I also notice that when running the build script, the `"kibana": {"version": "7.10.0"}` will be added automatically to `package.json` in the artifact. This is the only reason I preserved it. Anyway, I can remove it.

Comment (ftianli-amzn) [reply package.json]: I removed the Kibana version in `package.json`, at least people don't need to change too many versions during version upgrading.

File: package.json (hunk 1 above)
Comment (ftianli-amzn) [reply package.json:16]: Adding the `yarn` prefix is required to run a Javascript file, and the `plugin-helpers` is not an executable file now but a JS file. (For others' reference: https://yarnpkg.com/cli/run https://classic.yarnpkg.com/en/docs/package-json/#toc-scripts)

File: package.json (hunk 2 above)
Comment (ftianli-amzn) [reply package.json:39]: Looks like this is the only solution to deal with the new requirement for versions.

**Comment by ftianli-amzn** [comment]
I will rename the build artifact in `build` script, instead of in the CI workflow for consistency, so there is one more change.

**Review by dbbaughe — APPROVED** [review]

**Review by lezzago — APPROVED** [review]
